### PR TITLE
[Console] Fix line wrapping for decorated text in block output

### PR DESCRIPTION
**Pull request by grasmash**
| Q             | A
| ------------- | ---
| Branch?       | 
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| License       | MIT

Fixed bug that caused decorated text to be wrapped too early in SymfonyStyle->block().

### CODE REVIEW

**Comment by grasmash** [comment]
Quick before and after:
```
- // Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et             \n
- // dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea       \n
- // commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla     \n
- // pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim            \n
- // id est laborum                                                                                                      \n
+ // Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore      \n
+ // magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo      \n
+ // consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla             \n
+ // pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id         \n
+ // est laborum                                                                                                         \n
```

Notice that after, we are correctly filling the line with text up to 120 chars as expected. Before, the words were wrapped too early.

**Comment by stof** [comment]
what if there is extra decoration inside the message ?

**Comment by grasmash** [comment]
@stof 

> what if there is extra decoration inside the message ? 

Not sure what you mean. We're measuring the "decoration" with `Helper::strlen($message) - Helper::strlenWithoutDecoration($this->getFormatter(), $message);`. What other decoration might there be?

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 1]
```diff
@@ -466,17 +466,22 @@ private function createBlock(iterable $messages, string $type = null, string $st
         }
 
         // wrap and add newlines for each element
+        $message_lines_to_add = [];
         foreach ($messages as $key => $message) {
             if ($escape) {
                 $message = OutputFormatter::escape($message);
             }
 
-            $lines = array_merge($lines, explode(\PHP_EOL, wordwrap($message, $this->lineLength - $prefixLength - $indentLength, \PHP_EOL, true)));
```
Comment (grasmash) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php:474]: See https://github.com/kalessil/phpinspectionsea/blob/master/docs/performance.md#slow-array-function-used-in-loop

**Comment by grasmash** [comment]
Failing tests do not appear to be related to the code change.

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 2]
```diff
@@ -488,10 +495,12 @@ private function createBlock(iterable $messages, string $type = null, string $st
         foreach ($lines as $i => &$line) {
             if (null !== $type) {
                 $line = $firstLineIndex === $i ? $type.$line : $lineIndentation.$line;
-            }
+                }
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: should be reverted

File: src/Symfony/Component/Console/Style/SymfonyStyle.php [hunk 3]
```diff
@@ -466,17 +466,24 @@ private function createBlock(iterable $messages, string $type = null, string $st
         }
 
         // wrap and add newlines for each element
+        $messagesLines = [];
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Style/SymfonyStyle.php]: why this variable? can't we append to `$lines` directly?

Comment (grasmash) [reply src/Symfony/Component/Console/Style/SymfonyStyle.php]: Yes, good point.

**Comment by grasmash** [comment]
I'd like to note that while this PR improves the situation, it's still not perfect. The correct solution would actually be to follow the example of formatAndWrap() and actually parse the string one character at a time to wrap the line while not breaking a tag. 

I looked into using formatAndWrap(), but it's problematic. We really need the wrap aspect of it and not the format aspect.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @grasmash.

**Pull request merged**